---
title: "NeurIPS 2023: Single-Cell Perturbation Prediction"
subtitle: Generalizing experimental interventions to unseen contexts, a NeurIPS 2023 competition
description: |
  [Get Started on Kaggle](https://www.kaggle.com/competitions/open-problems-single-cell-perturbations){class="btn btn-primary"}
date: "2023-08-30"
start-date: "2023-09-11"
end-date: "2023-11-30"
title-block-banner-color: white
title-block-banner: ../../images/heros/competition_dark.jpg
page-layout: full
css: ../events.css
categories:
  - competition
image: image.png
---

:::{layout="[[30,70]]"}
::::{.section-header}
## Summary
::::

::::{.section-content}

Single-cell sequencing technologies have revolutionized our understanding of the heterogeneity and dynamics of cells and tissues. However, single-cell data analysis faces challenges such as high dimensionality, sparsity, noise, and limited ground truth. In this 3rd installment in the Open Problems in Single-Cell Analysis competitions at NeurIPS, we challenge competitors to develop algorithms capable of predicting single-cell perturbation response across experimental conditions and cell types. We will provide a new benchmark dataset of human peripheral blood cells under chemical perturbations, which simulate drug discovery experiments. The objective is to develop methods that can generalize to unseen perturbations and cell types to enable scientists to overcome the practical and economic limitations of single-cell perturbation studies. The goal of this competition is to leverage advances in representation learning (in particular, self-supervised, multi-view, and transfer learning) to unlock new capabilities bridging data science, machine learning, and computational biology. We hope this effort will continue to foster collaboration between the computational biology and machine learning communities to advance the development of algorithms for biomedical data.

::::
:::


:::{layout="[[30,70]]"}
::::{.section-header}
## Sponsors
::::

::::{.section-content}

```{=html}
<div class="grid">
  <div class="card card-shadow g-col-12 g-col-lg-5 g-start-lg-2">
    <img class="card-img-top" src="/images/sponsor/logotypemarkcolor.svg" alt="Card image cap" style="width:200px;  margin: 40px 50px 20px; background-color:white;">
    <div class="card-body" style="padding-top: 0px;">
      <p class="card-text">Cellarity is redefining drug discovery by targeting cell behaviors as opposed to individual proteins.<br><a class="stretched-link" href="https://cellarity.com/careers?utm_source=neurips&utm_medium=push-notification&utm_campaign=neurips&utm_content=neurips-competition">Learn more.</a></p>
    </div>
  </div>

  <div class="card card-shadow g-col-12 g-col-lg-5">
    <img class="card-img-top" src="/images/sponsor/CZI_Logotype.svg" style="width:150px; margin: 20px 60px 10px; background-color:white;">
    <div class="card-body" style="padding-top: 0px;">
      <p class="card-text" style="line-height: 1.3rem;">CZI leverages technology and collaboration to accelerate progress in science, education and community work.<br> <a class="stretched-link" href="https://czi.org">Learn more.</a></p>
    </div>
  </div>




  <div class="card card-shadow g-col-12 g-col-lg-5 g-start-lg-2">
    <img class="card-img-top" src="/images/sponsor/yale-logo-sprite.svg" alt="Card image cap" style="width:130px;  margin: 22px 60px;">
    <div class="card-body" style="padding-top: 0px;">
      <a class="stretched-link" href="https://www.yale.edu/"></a>
    </div>
  </div>




  <div class="card card-shadow g-col-12 g-col-lg-5">
    <img class="card-img-top" src="/images/sponsor/helmholtz_logotype.svg" alt="Card image cap" style="width:225px;  margin: 40px 12.5px">
    <div class="card-body" style="padding-top: 0px;">
      <a class="stretched-link" href="https://www.helmholtz-muenchen.de/icb/index.html"></a>
    </div>
  </div>
</div>

```
::::
:::

:::{layout="[[30,70]]"}
::::{.section-header}
## Organizers
::::

::::{.section-content}

_In alphabetical order_

-   **Daniel Burkhardt** (<a href="mailto:dburkhardt@cellarity.com?subject=%5BNeurIPS_2021%5D" style="color: #EB5252;">Primary contact</a>) is a Machine Learning Scientist at Cellarity, a biotechnology company in Boston. He is a core organizer of the Open Problems in Single-Cell Analysis project. He completed his PhD in Genetics at Yale University with a specialization in machine learning under the supervision of Smita Krishnaswamy. His dissertation focused on modeling experimental perturbations and biological heterogeneity in single-cell datasets. Daniel is also the core organizer of the Machine Learning for Single Cell Analysis workshop, offered bi-annually since 2019 by the Krishnaswamy Lab at Yale. 

-   **Robrecht Cannoodt** is a computer science engineer at Data Intuitive. He completed his PhD in Saeys lab at Ghent University, which was mainly focused on unsupervised learning in single-cell omics, more specifically on trajectory inference. In OpenProblems, Robrecht oversees infrastructure development for building collaborative and modular pipelines using Nextflow and Viash.

-   **Scott Gigante** is an Associate Staff Machine Learning Scientist at Immunai, a biotechnology company in New York. He is a core organizer of the Open Problems in Single-Cell Analysis project. He completed his PhD in Computational Biology at Yale University with a specialization in machine learning under the supervision of Smita Krishnaswamy and Ronald Coifman. Scott is also the core maintainer of the Open Problems for Single Cell Analysis living benchmark, and is focused on maximizing transferability of solutions from competitions to benefit the broader community.

-   **Christopher Lance** is a PhD candidate in the Machine Learning Group of Prof. Fabian Theis at the Helmholtz Center Munich. He graduated in Molecular Biotechnology at the University of Heidelberg with a focus on bioinformatics. In prior research projects at the Sanger Institute, UK and the EMBL Heidelberg, he worked on association studies of rare genetic variants with plasma proteomics and leveraging metabolomics data to characterize drug perturbations on bacterial metabolism. He joined the OpenProblems team during the first NeurIPS competition in 2021 on multimodal single cell data integration as one of the main data analysts and was in charge of the follow up assessment of the competition results and feedback, published in PMLR. He is now focusing on best practices for analysing multimodal single cell data and the integration of multiple modalities. Christopher will contribute to the analysis of the newly generated data for this years competition and is evaluating baseline methods and metrics.

-   **Malte LÃ¼cken** is a Principal Investigator at Helmholtz Munich and a core organizer of the Open Problems in Single-Cell Analysis project. His research focuses on building single-cell reference atlases, benchmarking computational methods for single-cell analysis, and investigating how environmental stimuli and natural variation manifests on the level of single cells. He has experience organizing hackathons including for the Single Cell Omics Germany network, Helmholtz Munich, and the Open Problems project, which provides the context for this proposal. Malte will be responsible for supervising the competition and shaping the dataset and metrics to answer key questions in perturbation biology.


-   **Angela Pisco** is the Director for Computational Biology at insitro, a biotechnology company in South San Francisco, and a core organizer of the Open Problems in Single-Cell Analysis project. Her main research interests are single cell genomics with a focus on building single cell atlas to understand health and disease. Angela's team is passionate about extracting meaningful information from biomedical datasets and use that to improve disease understanding and drug development.

::::
:::
