---
title: "NeurIPS 2022: Multimodal Single-Cell Integration Across Time, Individuals, and Batches"
subtitle: A competition on predicting multimodal expression across batches and timepoints
description: |
  A NeurIPS Competition (2022)

  [See on Kaggle](https://www.kaggle.com/competitions/open-problems-multimodal){class="btn btn-primary" style="border: 1px solid white;"}
date: "2022-08-15"
start-date: "2022-08-15"
end-date: "2022-11-15"
title-block-banner-color: white
title-block-banner: ../../images/heros/competition_dark.jpg
page-layout: full
css: ../events.css
categories:
  - competition
image: image.png
---

:::{layout="[[30,70]]"}
::::{.section-header}
## Summary
::::

::::{.section-content}

Scaling from a dozen cells a decade ago to millions of cells today, single-cell measurement technologies are driving a revolution in the life sciences. Recent advances make it possible to measure multiple high-dimensional modalities (e.g. DNA accessibility, RNA, and proteins) simultaneously in the same cell. Such data provides, for the first time, a direct and comprehensive view into the layers of gene regulation that drive biological diversity and disease.

In [2021](https://openproblems.bio/neurips_2021), we organized the first single-cell analysis competition at NeurIPS bringing together 280 participants to compete on an atlas-scale dataset of human bone marrow cells from 12 donors generated across 4 sites globally. 

In this competition for [NeurIPS 2022](https://neurips.cc/Conferences/2022/CompetitionTrack), we are extending the challenge to drive innovation in modeling temporal single-cell data measured in multiple modalities at multiple time points. In this years competition, we generated a 300,000-cell time course dataset of CD34+ hematopoietic stem and progenitor cells (HSPC) from four human donors at five time points. HSPCs are stem cells that give rise to all other cells in the blood throughout adult life, and a 10-day time course captures important biology in CD34+ HSPCs.

In the test set, taken from an unseen later time point in the dataset, competitors will be provided with one modality and be tasked with predicting a paired modality measured in the same cell. The added challenge of this competition is that the test data will be from a later time point than any time point in the training data.

To read all the details , [See on Kaggle!](https://www.kaggle.com/competitions/open-problems-multimodal/)
::::
:::


:::{layout="[[30,70]]"}
::::{.section-header}
## Sponsors
::::

::::{.section-content}

```{=html}
<div class="grid">
  <div class="card card-shadow g-col-12 g-col-lg-5 g-start-lg-2">
    <img class="card-img-top" src="/images/sponsor/logotypemarkcolor.svg" alt="Card image cap" style="width:200px;  margin: 40px 50px 20px; background-color:white;">
    <div class="card-body" style="padding-top: 0px;">
      <p class="card-text">Cellarity is redefining drug discovery by targeting cell behaviors as opposed to individual proteins.<br><a class="stretched-link" href="https://cellarity.com/careers?utm_source=neurips&utm_medium=push-notification&utm_campaign=neurips&utm_content=neurips-competition">Learn more.</a></p>
    </div>
  </div>

  <div class="card card-shadow g-col-12 g-col-lg-5">
    <img class="card-img-top" src="/images/sponsor/CZI_Logotype.svg" style="width:150px; margin: 20px 60px 10px; background-color:white;">
    <div class="card-body" style="padding-top: 0px;">
      <p class="card-text" style="line-height: 1.3rem;">CZI leverages technology and collaboration to accelerate progress in science, education and community work.<br> <a class="stretched-link" href="https://czi.org">Learn more.</a></p>
    </div>
  </div>



  <div class="card card-shadow g-col-12 g-col-lg-4">
    <img class="card-img-top" src="/images/sponsor/biohub_logo.svg" alt="Card image cap" style="width:170px;  margin: 25px 40px;">
    <div class="card-body" style="padding-top: 0px;">
      <a class="stretched-link" href="https://www.czbiohub.org/"></a>
    </div>
  </div>


  <div class="card card-shadow g-col-12 g-col-lg-4">
    <img class="card-img-top" src="/images/sponsor/yale-logo-sprite.svg" alt="Card image cap" style="width:130px;  margin: 22px 60px;">
    <div class="card-body" style="padding-top: 0px;">
      <a class="stretched-link" href="https://www.yale.edu/"></a>
    </div>
  </div>




  <div class="card card-shadow g-col-12 g-col-lg-4">
    <img class="card-img-top" src="/images/sponsor/helmholtz_logotype.svg" alt="Card image cap" style="width:225px;  margin: 40px 12.5px">
    <div class="card-body" style="padding-top: 0px;">
      <a class="stretched-link" href="https://www.helmholtz-muenchen.de/icb/index.html"></a>
    </div>
  </div>
</div>

```
::::
:::

:::{layout="[[30,70]]"}
::::{.section-header}
## Organizers
::::

::::{.section-content}

_In alphabetical order_

-   **Daniel Burkhardt** (<a href="mailto:dburkhardt@cellarity.com?subject=%5BNeurIPS_2021%5D" style="color: #EB5252;">Primary contact</a>) is a Machine Learning
    Scientist at [Cellarity.](https://cellarity.com) He is a core organizer of the Open Problems
    in Single-Cell Analysis. Daniel completed his PhD in Genetics at
    Yale University with a specialization in machine learning under the
    supervision of Smita Krishnaswamy.

-   **Jonathan M. Bloom** leads Perturbation Biology and Machine Learning at Cellarity. As an Institute Scientist at the Broad Institute, he cofounded the Hail team, Learning Meaningful Representations of Life, and Models, Inference & Algorithms, while contributing to sequencing benchmarks, statistical genetics, ML theory, and neuroscience efforts. Jonathan completed his PhD at Columbia and Moore Instructorship and NSF postdoc at MIT, conducting research on algebraic topology and geometry while reimagining the teaching of statistics.  

-   **Robrecht Cannoodt** is a postdoctoral researcher in Saeys lab at 
    VIB-UGent and a computer science consultant at Data Intuitive.
    During his PhD, his research was focused mainly on unsupervised learning in 
    single-cell omics, more specifically on trajectory inference. Robrecht 
    oversees infrastructure development for building collaborative, scalable 
    and reproducible pipelines using NextFlow and Viash.
    
-   **Peter Holderrieth** is a Machine Learning Scientist at Cellarity where he works on hit 
    prediction tools based on single-cell data. Previously, he was an Associate Scientist at 
    Genomics plc. Originally trained as a mathematician, he holds an MSc in Statistics and 
    MSc in Neuroscience from the University of Oxford.
    

-   **Smita Krishnaswamy** is an associate professor of Computer Science and
    Genetics at Yale University. Her research focuses on unsupervised
    learning, using data geometry, topology and deep learning methods
    for big biomedical data. She is a faculty advisor for Open Problems
    in Single-Cell Analysis.

-   **Christopher Lance** is a PhD candidate in the Machine Learning Group of Prof. Fabian Theis 
    at the Helmholtz Center Munich where he works on establishing best practices for analysing multimodal single-cell data
    and the integration of multiple modalities.

-   **Malte Lücken** is a senior postdoctoral researcher and team leader
    for integrative single-cell analysis in the Machine Learning Group
    of Prof. Fabian Theis. His research focuses on evaluating
    computational methods for single-cell analysis and investigating how
    environmental stimuli and natural variation manifests on the level
    of single cells.


-   **Angela Pisco** is the Associate Director of Bioinformatics at the Chan Zuckerberg Biohub.
    Her group is responsible for generating and annotated atlas-scale single-cell
    datasets across biological systems.

-   **Fabian Theis** is the Professor of Mathematical Modeling of Biological
    Systems at the Department of Mathematics of the Technical University
    of Munich; the director of the Institute of Computational Biology,
    Helmholtz Zentrum München; the scientific director of the Helmholtz
    Artificial Intelligence Cooperation Unit; and a faculty member of
    the Wellcome Trust Sanger Institute, Cambridge, UK. His lab develops innovative
    methods for single-cell analysis.

::::
:::