{
  "cells": [
    {
      "cell_type": "raw",
      "metadata": {},
      "source": [
        "---\n",
        "title: Design the API\n",
        "order: 20\n",
        "---"
      ],
      "id": "503f8575"
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "# API file formats\n",
        "\n",
        "\n",
        "\n",
        "```{yaml}\n",
        "type: file # <1>\n",
        "description: \"A normalised data with a PCA embedding, HVG selection and a kNN graph\" # <2>\n",
        "example: \"dataset.h5ad\"  # <3>\n",
        "info:  # <4>\n",
        "  label: \"Dataset+PCA+HVG+kNN\"\n",
        "  slots:\n",
        "    # \n",
        "\n",
        "```\n",
        "\n",
        "1. Type of the entry. For file speicifications, it is always `file`\n",
        "2. Description of the file, useful for quickly understanding what the file contains\n",
        "3. Example of the file name in the command line\n",
        "4. Info\n",
        "\n",
        "\n",
        "### Example\n",
        "\n",
        "```{yaml}\n",
        "type: file\n",
        "description: \"A normalised data with a PCA embedding, HVG selection and a kNN graph\"\n",
        "example: \"dataset.h5ad\"\n",
        "info:\n",
        "  label: \"Dataset+PCA+HVG+kNN\"\n",
        "  slots:\n",
        "    layers: \n",
        "      - type: integer\n",
        "        name: counts\n",
        "        description: Raw counts\n",
        "        required: true\n",
        "      - type: double\n",
        "        name: normalized\n",
        "        description: Normalised expression values\n",
        "    obs:\n",
        "      - type: string\n",
        "        name: celltype\n",
        "        description: Cell type information\n",
        "        required: false\n",
        "    var:\n",
        "      - type: boolean\n",
        "        name: hvg\n",
        "        description: Whether or not the feature is considered to be a 'highly variable gene'\n",
        "        required: true\n",
        "    uns:\n",
        "      - type: string\n",
        "        name: dataset_id\n",
        "        description: \"A unique identifier for the dataset\"\n",
        "        required: true\n",
        "```\n",
        "\n",
        "# API component specs\n",
        "\n",
        "\n",
        "{{< include ../_blocks/_clone_repo.qmd >}}\n",
        "{{< include /_include/_synchonize_tabs.qmd >}}\n",
        "\n",
        "Components are the most modular building blocks of the overall task.\n",
        "Each method, metric or data processing workflow should have its own component.\n",
        "This way, components can be separated arranged into end-to-end workflows.\n",
        "\n",
        "For different variants of the same task, such as multiple different methods or metrics, we can define a common API so that the further processing of the outputs can be kept the same across methods or metrics.\n",
        "It is good practice to formalise the common metadata of all methods or metrics in a generic API specification.\n",
        "\n",
        "## Specification\n",
        "\n",
        "madtory vs optional vs extendable\n",
        "\n",
        "<!-- TODO: generate table with different  -->\n",
        "\n",
        "```{yaml}\n",
        "functionality:  # <1>\n",
        "  namespace: <task_name>/<subtask>\n",
        "  info:\n",
        "    type: method\n",
        "    output_type: embedding\n",
        "  arguments:\n",
        "    - name: --input\n",
        "      __merge__: anndata_unintegrated.yaml\n",
        "    - name: --output\n",
        "      direction: output\n",
        "      __merge__: anndata_integrated_embedding.yaml\n",
        "    - name: --hvg\n",
        "      type: boolean\n",
        "      description: Whether to subset to highly variable genes\n",
        "      default: false\n",
        "      required: false\n",
        "  test_resources:\n",
        "    - path: ../../../../resources_test/batch_integration/pancreas/\n",
        "    - type: python_script\n",
        "      path: generic_test.py\n",
        "```\n",
        "\n",
        "### Example\n",
        "\n",
        "```{yaml}\n",
        "functionality:\n",
        "  namespace: <task_name>/<subtask>\n",
        "  info:\n",
        "    type: method\n",
        "    output_type: embedding\n",
        "  arguments:\n",
        "    - name: --input\n",
        "      __merge__: anndata_unintegrated.yaml\n",
        "    - name: --output\n",
        "      direction: output\n",
        "      __merge__: anndata_integrated_embedding.yaml\n",
        "    - name: --hvg\n",
        "      type: boolean\n",
        "      description: Whether to subset to highly variable genes\n",
        "      default: false\n",
        "      required: false\n",
        "  test_resources:\n",
        "    - path: ../../../../resources_test/batch_integration/pancreas/\n",
        "    - type: python_script\n",
        "      path: generic_test.py\n",
        "```\n",
        "\n",
        "\n",
        "## Applying API to specific component\n",
        "\n",
        "```{yaml}\n",
        "__merge__: ../../api/comp_method_embedding.yaml\n",
        "functionality:\n",
        "  name: scanorama_embed\n",
        "  description: \"Efficient integration of heterogeneous single-cell \n",
        "    transcriptomes using Scanorama\"\n",
        "  info:\n",
        "    method_name: Scanorama\n",
        "    paper_reference: \"hie2019efficient\"\n",
        "    code_url: https://github.com/brianhie/scanorama\n",
        "    v1_url: openproblems/tasks/_batch_integration/batch_integration_graph/methods/scanorama.py\n",
        "    v1_commit: 29803b95c88b4ec5921df2eec7111fd5d1a95daf\n",
        "    preferred_normalization: log_cpm\n",
        "    variants:\n",
        "      scanorama_embed_full_unscaled:\n",
        "      scanorama_embed_hvg_unscaled:\n",
        "        hvg: true\n",
        "      scanorama_embed_hvg_scaled:\n",
        "        hvg: true\n",
        "        preferred_normalization: log_cpm_scaled\n",
        "      scanorama_embed_full_scaled:\n",
        "        preferred_normalization: log_cpm_scaled\n",
        "  resources:\n",
        "    - type: python_script\n",
        "      path: script.py\n",
        "platforms:\n",
        "  - type: docker\n",
        "    image: mumichae/scib-base:1.1.3\n",
        "    setup:\n",
        "      - type: python\n",
        "        pypi:\n",
        "          - scanorama\n",
        "          - pyyaml\n",
        "  - type: nextflow\n",
        "```"
      ],
      "id": "7c3d676f"
    }
  ],
  "metadata": {
    "kernelspec": {
      "name": "python3",
      "language": "python",
      "display_name": "Python 3 (ipykernel)"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}